vortex-datafusion/src/persistent/execution.rs

```diff
@@ -162,6 +166,26 @@ impl ExecutionPlan for VortexExec {
     }
 }
 
+fn make_vortex_predicate(
+    predicate: Option<Arc<dyn PhysicalExpr>>,
+    dtype: DType,
+) -> VortexResult<Option<Arc<dyn VortexExpr>>> {
+    predicate
+        .as_ref()
+        // If we cannot convert an expr to a vortex expr, we run no filter, since datafusion
+        // will rerun the filter expression anyway.
+        .and_then(|expr| {
+            // This splits expressions into conjunctions and converts them to vortex expressions.
+            // Any inconvertible expressions are dropped since true /\ a == a.
+            datafusion_physical_expr::split_conjunction(expr)
+                .into_iter()
+                .filter_map(|e| convert_expr_to_vortex(e.clone()).ok())
+                .reduce(and)
+                .map(|expr| simplify_typed(expr, &dtype))
```

Comment:
Do we have to do this ourselves? Shouldn't the scan do it? Otherwise it's quite a fragile API

Reply:
removed, trusting the scan